sql/org_mozilla_vrbrowser_derived/clients_daily_v1/query.sql

```diff
@@ -0,0 +1,169 @@
+CREATE TEMP FUNCTION udf_geo_struct(
+  country STRING,
+  city STRING,
+  geo_subdivision1 STRING,
+  geo_subdivision2 STRING
+) AS ( IF(
+    country IS NULL
+    OR country = '??',
+    NULL,
+    STRUCT(
+      country,
+      NULLIF(city, '??') AS city,
+      NULLIF(geo_subdivision1, '??') AS geo_subdivision1,
+      NULLIF(geo_subdivision2, '??') AS geo_subdivision2
+    )
+  )
+);
+
+CREATE TEMP FUNCTION
+  udf_glean_timespan_seconds(timespan STRUCT<time_unit STRING, value INT64>)
+  RETURNS INT64 AS (
+    CAST(
+      FLOOR(
+        CASE timespan.time_unit
+          WHEN 'nanosecond' THEN timespan.value / 1000 / 1000 / 1000
+          WHEN 'microsecond' THEN timespan.value / 1000 / 1000
+          WHEN 'millisecond' THEN timespan.value / 1000
+          WHEN 'second' THEN timespan.value
+          WHEN 'minute' THEN timespan.value * 60
+          WHEN 'hour' THEN timespan.value * 60 * 60
+          WHEN 'day' THEN timespan.value * 60 * 60 * 24
+        END )
+      AS INT64));
+
+CREATE TEMP FUNCTION udf_json_mode_last(list ANY TYPE) AS (
+  (
+    SELECT
+      ANY_VALUE(_value)
+    FROM
+      UNNEST(list) AS _value
+    WITH OFFSET AS _offset
+    GROUP BY
+      TO_JSON_STRING(_value)
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1
+  )
+);
+
+CREATE TEMP FUNCTION
+  udf_mode_last(list ANY TYPE) AS ((
+    SELECT
+      _value
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
+    GROUP BY
+      _value
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1 ));
+
+WITH unioned AS (
+  SELECT
+    submission_timestamp,
+    document_id,
+    client_info,
+    sample_id,
+    metadata,
+    normalized_channel,
+    metrics AS baseline_metrics,
+    NULL AS metrics
+  FROM
+    org_mozilla_vrbrowser_stable.baseline_v1
+  UNION ALL
+  SELECT
+    submission_timestamp,
+    document_id,
+    client_info,
+    sample_id,
+    metadata,
+    normalized_channel,
+    NULL AS baseline_metrics,
+    metrics
+  FROM
+    org_mozilla_vrbrowser_stable.metrics_v1
+),
+  --
+base AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    LOWER(client_info.client_id) AS client_id,
+    *
+  FROM
+    unioned
+  WHERE
+    client_info.client_id IS NOT NULL
+),
+  --
+windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    'Firefox Reality' AS app_name,
```

Comment:
Is there a list with these names somewhere?

Reply:
There's a partial list in https://docs.telemetry.mozilla.org/concepts/choosing_a_dataset_mobile.html#products-overview but it's kind of a judgment call right now.

We should definitely have a more coherent and well-documented story for this as part of generalizing derived tables for Glean apps.